packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2001,12 +2418,11 @@ export class TransactionController extends BaseControllerV1<
   private async approveTransaction(transactionId: string) {
     const { transactions } = this.state;
     const releaseLock = await this.mutex.acquire();
-    const chainId = this.getChainId();
     const index = transactions.findIndex(({ id }) => transactionId === id);
     const transactionMeta = transactions[index];
-
     const {
       txParams: { from },
+      networkClientId,
```

Comment:
Perhaps an illogical edge case, but suppose we create a transaction approval, then delete the network client, and then approve.

Would that state be detected from the `NetworkController` and throw nicely or reject the approvals, or would it fail when we next retrieve the client?

Reply:
> Perhaps an illogical edge case, but suppose we create a transaction approval, then delete the network client, and then approve.

I'd imagine this isn't possible. But I suppose just to be sure this might favor the approach I was advocating of deleting all unsubmitted transactions for a networkClient that is removed cc @jiexi

Reply:
Scenario: networkClient for a transaction ceases to exist before submission but after approval, multichain flag is on

Result: Currently will attempt to fallback to another networkClient with the same chainId. If none, then use the global provider which is highly likely to be incorrect. If this is a massive issue, we may need to update the fallback logic of #getEthQuery

Reply:
> Scenario: networkClient for a transaction ceases to exist before submission but after approval, multichain flag is on

actually I remember now we did test and this isn't currently possible. But it might be possible in Amon Hen UX.